### DISCUSSION TITLE
Feat: Use sso_start_url from AWS Profile

### DISCUSSION THREAD

**Discussion by maschnetwork** [aws/codewhisperer-command-line-discussions#71] [Feature request]
### Checks

- [X] I could not find an existing setting that solved my issue
- [X] I searched [the discussions in this repo](https://github.com/aws/codewhisperer-command-line-discussions/discussions?q=) and there are no duplicates of my feature request

### Description:

Currently the cli prompts for the start url when using Identity center login. It would be great if the sso_start_url / current session of the configured AWS profile can be used.

**Comment by brendanfalk** [comment]
Great idea. Just to clarify, is your configured AWS profile saved locally or in your cookies in the browser? Both could work

**Reply by maschnetwork** [reply]
It is locally within the config file: https://docs.aws.amazon.com/cli/latest/userguide/sso-configure-profile-token.html#sso-configure-profile-token-manual

**Reply by grant0417** [reply]
Do you want us just to pull from the `default` profile?

**Reply by maschnetwork** [reply]
Ideally the experience matches the aws cli - so I would be able to pass the profile as well.

**Reply by grant0417** [reply]
So the flow is something like this:

- If no SSO profiles: 
	- Current flow, Start URL and Region
- If a SSO profile exists
	- Show the profile options in a dropdown OR (Start URL and region)

**Reply by brendanfalk** [reply]
@maschnetwork this is an awesome idea. Do you have any insight into how many people actually put their start url in their AWS config?

**Reply by maschnetwork** [reply]
@grant0417 @brendanfalk not sure if a drop down is needed when you have the possibility to pass --profile as well. Maybe both to be flexible. I don't have any numbers on this but this is a very common way to configure it.

**Reply by bneqld** [reply]
Hi everybody,
not sure if I understand all of your comments. I have an SSO profile in my aws config and have been trying to use that directly for cw login from the command line. Currently I always to a SSO (which I do not actually need) just to get my login URL printed, which I can then reuse for the cw login. It would be nice to be able to just give a profile in my aws config as an option for cw login!

**Reply by brendanfalk** [reply]
@bneqld Agree. We will prioritize adding. Could you please upvote this post for us?

Also, you shouldn't have to authenticate that often... If you are, something is definitely going wrong. How often are you being asked to auth?
